### ISSUE TITLE
Add rule(s) to check for usage of deprecated, removed, and/or old versions of APIs

### ISSUE THREAD

**Issue by alrra** [webhintio/hint#30]

**Comment by alrra** [comment]
@dstorey We had a discussion about this the other day, can you provide more details here? Thanks!

**Comment by dstorey** [comment]
I'm putting together more details on what could be done here.

**Comment by alrra** [comment]
Maybe this should be a different rule/issue, but I'm leaving it here for now: https://github.com/whatwg/html/issues/3076

**Comment by antross** [comment]
We could also expand the scope of this hint to notify users about new APIs which aren't yet available somewhere within their chosen browser support matrix (see #1203 for CSS) and HTML deprecations (see #1204).

We just had a good conversation today with @digitarald and @martinbalfanz from Mozilla and it sounds like the [MDN compatibility data](https://github.com/mdn/browser-compat-data/) would be a good fit to help us build this hint. There's also a Firefox/Chrome DevTools extension called [Compat Report](https://github.com/eduardoboucas/compat-report) which is built on this data we could look at as an example. Currently the MDN data is complete for CSS with work in-progress to finish the rest of the platform APIs.

I'm particularly interested in this space and would be happy to help us move forward on getting this hint started once I wrap up with the VS Code extension.

**Comment by dstorey** [comment]
I'd still like to see WebIDL used to keep the MDN data up to date as it is often incorrect (at least for Edge) or out of date (relies on someone to manually check each release and not miss anything). Having tooling that checks both WebIDL and type mirror to guide updating MDN (if it isn't allowed to be automatic) would be fantastic. E.g. Google's type mirror tool that compares with MDN: https://mdittmer.github.io/mdn-confluence/multi.html and https://github.com/mdittmer/mdn-confluence . Both data sources have their uses as some data is lost with type mirror (including non enumerables, parameters, etc. while sometimes WebIDL may be incorrect (such as if something is off by default or a API is broken somehow)

**Comment by antross** [comment]
> I'd still like to see WebIDL used to keep the MDN data up to date

@dstorey Totally agree that figuring out how to use WebIDL to help keep whatever data source we use up-to-date is a good idea. I noticed there is an [open issue](https://github.com/mdn/browser-compat-data/issues/2306) on the MDN compat data repo about looking into WebIDL for keeping in sync with specifications. 

I'd be curious to have @patrickkettner's input too, but it seems like it would be worthwhile to start a discussion or new issue there about also using the WebIDL from browsers?

**Comment by dstorey** [comment]
I think so as it is a good source of thruth for the most part, with all browsers using it to generate code (with some caveats)

**Comment by digitarald** [comment]
> I'd still like to see WebIDL used to keep the MDN data up to date as it is often incorrect (at least for Edge) or out of date (relies on someone to manually check each release and not miss anything).

@dstorey is that still the case with the recent involvement from all browsers to keep the data fresh? I wonder if it is on webhint to figure this problem out, since the browser-compat-data seems to be working on improving correctness and reducing the update latency.

**Comment by patrickkettner** [comment]
it is substantially less true than it was previously, but it is still an issue. IDL code will almost always be more accurate than a manually curated list

**Comment by digitarald** [comment]
I can understand the risk of using bad data, its a GIGO risk and it can dissolve the trust in the audit. My expectation would be, and I am happy to take point on driving this, that MDN's data will keep aiming to be the most complete and up to date dataset.

As perfection will be never achieved with this kind of aggregated date, what level of accuracy do we expect in the data that powers the audits? Or more specific, what is good enough to unblock this discussion?

**Comment by molant** [comment]
Closing in favor of https://github.com/webhintio/rfcs/issues/10
